code/modules/flufftext/Hallucination.dm

```diff
@@ -346,82 +397,122 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 		if("esword")
+			iterations_left = rand(4, 8)
 			target.playsound_local(source, 'sound/weapons/saberon.ogg',15, 1)
-			for(var/i in 1 to rand(4, 8))
-				target.playsound_local(source, 'sound/weapons/blade1.ogg', 50, 1)
-				if(i == 4)
-					target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-				sleep(rand(CLICK_CD_MELEE, CLICK_CD_MELEE + 6))
-			target.playsound_local(source, 'sound/weapons/saberoff.ogg', 15, 1)
 		if("gun")
-			var/hits = 0
-			for(var/i in 1 to rand(3, 6))
-				target.playsound_local(source, 'sound/weapons/gun/shotgun/shot.ogg', 25, TRUE)
-				if(prob(60))
-					addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, 'sound/weapons/pierce.ogg', 25, 1), rand(5,10))
-					hits++
-				else
-					addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, "ricochet", 25, 1), rand(5,10))
-				sleep(rand(CLICK_CD_RANGE, CLICK_CD_RANGE + 6))
-				if(hits >= 2 && prob(80))
-					target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-					break
+			iterations_left = rand(3, 6)
 		if("stunprod") //Stunprod + cablecuff
+			process = FALSE
 			target.playsound_local(source, 'sound/weapons/egloves.ogg', 40, 1)
 			target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-			sleep(20)
-			target.playsound_local(source, 'sound/weapons/cablecuff.ogg', 15, 1)
+			addtimer(CALLBACK(target, /mob/.proc/playsound_local, source, 'sound/weapons/cablecuff.ogg', 15, 1), 20)
 		if("harmbaton") //zap n slap
+			iterations_left = rand(5, 12)
 			target.playsound_local(source, 'sound/weapons/egloves.ogg', 40, 1)
 			target.playsound_local(source, get_sfx("bodyfall"), 25, 1)
-			sleep(20)
-			for(var/i in 1 to rand(5, 12))
-				target.playsound_local(source, "swing_hit", 50, 1)
-				sleep(rand(CLICK_CD_MELEE, CLICK_CD_MELEE + 4))
+			next_action = 2 SECONDS
 		if("bomb") // Tick Tock
-			for(var/i in 1 to rand(3, 11))
+			iterations_left = rand(3, 11)
+
+	if (process)
+		START_PROCESSING(SSfastprocess, src)
+	else
+		qdel(src)
+
+/datum/hallucination/battle/process(delta_time)
+	next_action -= (delta_time * 10)
+
+	if (next_action <= 0)
+		switch (battle_type)
+			if ("disabler", "laser", "gun")
+				var/fire_sound
+				var/hit_person_sound
+				var/hit_wall_sound
+				var/number_of_hits
+				var/chance_to_fall
+
+				switch (battle_type)
+					if ("disabler")
+						fire_sound = 'sound/weapons/taser2.ogg'
+						hit_person_sound = 'sound/weapons/tap.ogg'
+						hit_wall_sound = 'sound/weapons/effects/searwall.ogg'
+						number_of_hits = 3
+						chance_to_fall = 70
+					if ("laser")
+						fire_sound = 'sound/weapons/laser.ogg'
+						hit_person_sound = 'sound/weapons/sear.ogg'
+						hit_wall_sound = 'sound/weapons/effects/searwall.ogg'
+						number_of_hits = 4
+						chance_to_fall = 70
+					if ("gun")
+						fire_sound = 'sound/weapons/gun/shotgun/shot.ogg'
+						hit_person_sound = 'sound/weapons/pierce.ogg'
+						hit_wall_sound = "ricochet"
+						number_of_hits = 2
+						chance_to_fall = 80
```

Comment:
Can you do this in a cleaner manner? This indentation gives me hives

Reply:
Any suggestions? 😛

Reply:
If I had any I'd make them 😞